File: ex2_calculo_preco_desconto/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Preço do produto: ");
        double preco = Double.parseDouble(scanner.nextLine());

        System.out.print("Percentual de desconto: ");
        double percentual = Double.parseDouble(scanner.nextLine());

        double desconto = preco * percentual / 100.0;
        double precoFinal = preco - desconto;

        System.out.printf("Desconto: R$ %.2f%n", desconto);
        System.out.printf("Preço final: R$ %.2f%n", precoFinal);

        scanner.close();
    }
}

File: ex2_calculo_preco_desconto/main.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    double preco, percentual;

    cout << "Preço do produto: ";
    cin >> preco;
    cout << "Percentual de desconto: ";
    cin >> percentual;

    double desconto = preco * percentual / 100.0;
    double precoFinal = preco - desconto;

    cout << fixed << setprecision(2);
    cout << "Desconto: R$ " << desconto << endl;
    cout << "Preço final: R$ " << precoFinal << endl;

    return 0;
}